language/tools/move-package/src/resolution/dependency_graph.rs

```diff
@@ -428,41 +468,53 @@ impl DependencyGraph {
                 );
             }
 
-            match self.package_table.entry(ext_name) {
-                Entry::Vacant(entry) => {
-                    entry.insert(ext_pkg);
-                }
-
-                // Seeing the same package in `extension` is OK only if it has the same set of
-                // dependencies as the existing one.i
-                Entry::Occupied(_) => {
-                    let (self_deps, ext_deps) =
-                        pkg_deps_equal(ext_name, &self.package_graph, &ext_graph);
-                    if self_deps != ext_deps {
-                        bail!(
-                            "Conflicting dependencies found for '{ext_name}' during external resolution by '{resolver}':\n{}{}",
-                            format_deps("\nExternal dependencies not found:", self_deps),
-                            format_deps("\nNew external dependencies:", ext_deps),
-                        );
+            let pkg = match self.package_table.get(&ext_name) {
+                None => ext_pkg,
+                Some(existing_pkg) => {
```

Comment:
nit: You should still be able to use the `entry` API for this -- `Entry::Occupied` contains the existing package.